src/judoscale/core/config.py

```diff
@@ -25,12 +23,17 @@ def for_report(self):
         }
 
     def _prepare_logging(self):
+        logger = logging.getLogger('judoscale')
```

Comment:
Can you help me understand the difference between this and `logging.getLogger(__name__)`? What is `__name__`? Is "judoscale" special in some way, or is this just an arbitrary name?

Reply:
`__name__` is a built-in attribute of Python that translates to the name of the file from which is called from. That's why sometimes at the end of a python file we see the expression `if __name__ == __main__: ` which means that if we run the file directly and do not import it as a module in another file, the __name__ translates to __main__ and then if that's the case we may want to do something particular or just call the functions and classes defined within that file. When we import the module in another file, this `__name__` translated to the path of the module.
I gave `judosacale` inside the getLogger because this is the library name, aka the root directory where django and core dirs are subdirectories. It is not just a random name. If you add the variable `name` in the formatter, which corresponds to `__name__`, you will see the names appear as `judoscale.core.adapter_api_client`, judoscale.django.apps` etc (it was the first screenshot I sent in the upwork messager). 
![judoscale_logging_with_name](https://user-images.githubusercontent.com/5695572/179212560-d7841c20-3f12-438d-8ae1-646a473dac5c.png)

About the logging, I read in the docs, that if you want to start a parent logger and then children loggers can attach to it , you need to strat the logger with the package/module name where it is called for the first time.

> Logging is performed by calling methods on instances of the [Logger](https://docs.python.org/3/library/logging.html#logging.Logger) class (hereafter called loggers). Each instance has a name, and they are conceptually arranged in a namespace hierarchy using dots (periods) as separators. For example, a logger named ‘scan’ is the parent of loggers ‘scan.text’, ‘scan.html’ and ‘scan.pdf’. Logger names can be anything you want, and indicate the area of an application in which a logged message originates.

https://docs.python.org/3/howto/logging.html#logging-advanced-tutorial

Reply:
This is very helpful. Thanks!